packages/x-date-pickers/src/internals/hooks/usePickerState.ts

```diff
@@ -21,18 +23,54 @@ export interface PickerStateValueManager<TInputValue, TDateValue> {
 
 export type PickerSelectionState = 'partial' | 'shallow' | 'finish';
 
-interface Draftable<T> {
-  committed: T;
+interface DateState<T> {
+  /**
+   * Date internally used on the picker and displayed in the input.
+   * It is updates whenever the user validates a step.
+   */
   draft: T;
+  /**
+   * Last full date provided by the user
+   * Is not updated when validating a step of a multistep picker (e.g. validating the date of a date time picker)
+   */
+  committed: T;
+  /**
+   * Date that will be used if the pickers tries to reset its value
+   */
+  resetTarget: T;
 }
 
-interface DraftAction<DraftValue> {
-  type: 'update' | 'reset';
-  payload: DraftValue;
+type DateStateActionType =
+  /**
+   * Set the draft, committed and accepted dates to the action value
+   * Closes the picker
+   */
+  | 'acceptAndClose'
+  /**
+   * Set the draft, committed and accepted dates to the action value
+   */
+  | 'setAll'
+  /**
+   * Set the draft and committed date to the action value
+   */
+  | 'setCommitted'
+  /**
+   * Set the draft date to the action value
+   */
+  | 'setDraft';
+
+interface DateStateAction<DraftValue> {
+  action: DateStateActionType;
+  value: DraftValue;
+  /**
+   * If `true`, do not fire the `onChange` callback
+   * @default false
+   */
+  internal?: boolean;
```

Comment:
`internal` has no meaning.

```suggestion
  skipOnChangeCall?: boolean;
```